client/get_approved_session/src/main.rs

```diff
@@ -16,9 +17,12 @@ const ARG_TOKEN_ID: &str = "token_id";
 
 #[no_mangle]
 pub extern "C" fn call() {
-    let nft_contract_hash: ContractHash = runtime::get_named_arg(ARG_NFT_CONTRACT_HASH);
+    let nft_contract_hash: ContractHash = runtime::get_named_arg::<Key>(ARG_NFT_CONTRACT_HASH)
+        .into_hash()
+        .map(|hash| ContractHash::new(hash))
+        .unwrap();
     let key_name: String = runtime::get_named_arg(ARG_KEY_NAME);
-    let token_id = runtime::get_named_arg::<U256>(ARG_TOKEN_ID);
+    let token_id = runtime::get_named_arg::<String>(ARG_TOKEN_ID);
```

Comment:
instead of overloading a single string argument, I think you should have two different arguments; `token_id` and `token_hash` 

the logic would expect the appropriate strongly typed arg based upon the contract's configured `token_identifier_kind` mode

Reply:
Implemented two different args which are looked up based on the identifier mode